listenbrainz/mbid_mapping/mapping/mbid_mapping.py

```diff
@@ -23,15 +26,17 @@ def create_tables(mb_conn):
         with mb_conn.cursor() as curs:
             curs.execute("DROP TABLE IF EXISTS mapping.tmp_mbid_mapping")
             curs.execute("""CREATE TABLE mapping.tmp_mbid_mapping (
+                                         id                        SERIAL,
```

Comment:
SERIAL sets up a sequence and uses it to fill in this value if you don't provide it. You're generating your own ids and inserting them with `insert_rows(mb_curs2, "mapping.tmp_mbid_mapping", rows)`, which means that the sequence won't be advanced. I can't see you creating an index or PK on this field, it might be better to make it explicitly `INTEGER NOT NULL` so that we don't run into any unexpected problems later

Reply:
in fact, I see you doing `DELETE FROM mapping.tmp_mbid_mapping WHERE id IN (` so maybe an index is a good idea here too (unless I can't see where you're doing it)

Reply:
Hehehhehe, yes this part was a bit... odd. When inserting using execute_values, it isn't exactly clear (supported?) on how to define DEFAULT as one of the values, which should be passed instead of an INT for serial. But since I am building the table by hand and can easily add an increment, I'll just manually specify them.

As for actually needing an index on this column and the delete in question -- this is done before indexes exist -- its being done FOR index creation. I want a UNIQUE index on the table and thus need to remove dups that resolve to the same combined_lookup value -- dups are not needed and complicate querying.

I don't see needing that column/index post table creation/population, since all the lookups happen via combined_lookup.

Reply:
Yeah, I guessed that the reason for doing the serial manually was because you can't set a default without writing out all of the columns in the INSERT statement (`INSERT INTO table (recording_name, recording_mbid) VALUES (1, 2)` etc).

if the deduplicate process is fast enough, you probably don't need to bother with the index then. I guess that this will do a table scan instead of index lookup for the `WHERE id IN (...)` clause, though? Does this have the potential to slow down in the future if this table gets bigger?

My comment about SERIAL was that there is a potential data inconsistency error:
 - set id SERIAL, default value becomes `nextval(id sequence)`
 - do a bulk insert, setting your own id values
 - at some point in time, try and insert a row into the table, relying on the sequence to give you an id number
 - the next value from the sequence already exists in the table and fails

of course, the insert will only fail if you have a unique index on id, which you don't. Also,  you're only ever creating this table and its contents automatically, so you'll probably never run into this situation.

My only suggestion here was to recommend to set the column type explicitly to INTEGER instead of SERIAL to make it explicit that you're not relying on any magic sequences.

Reply:
ok, that last point is a good suggestion, I like it.